### PR TITLE
Use Leo TraceId

### PR DESCRIPTION
**Pull request by rtitle**
Note the target of this PR is https://github.com/DataBiosphere/leonardo/pull/1061/. See https://github.com/DataBiosphere/leonardo/pull/1061/#discussion_r324776435 for context. This makes Mockito happy by matching a non-value object. Feel free to use or not.

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala [hunk 1]
```diff
@@ -384,6 +384,8 @@ object ExtensionType extends Enum[ExtensionType] {
   case object LabExtension extends ExtensionType
 }
 
+case class TraceId(uuid: UUID)
```
Comment (Qi77Qi) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala:387]: just curious, why does using a separate data type fixes the null pointer exception?

Comment (rtitle) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala:387]: I think because it's not a value object

Comment (rtitle) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala:387]: Compare with https://github.com/broadinstitute/workbench-libs/blob/develop/model/src/main/scala/org/broadinstitute/dsde/workbench/model/models.scala#L6

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala:387]: sorry forgot to comment back here...I saw the other comment :)

**Comment by rtitle** [comment]
I see this was merged... @Qi77Qi came up with a better solution, probably want to back this out :)

**Pull request merged**